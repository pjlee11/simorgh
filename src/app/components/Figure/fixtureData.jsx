import React from 'react';
import FigureContainer from '../../containers/Figure';
import { ServiceContext } from '../ServiceContext';

const imageAlt = 'Pauline Clayton';
const imageSrc =
  'https://ichef.bbci.co.uk/news/640/cpsprodpb/E7DB/production/_101655395_paulineclayton.jpg';
const imageRatio = 56.25;
const captionValue = 'This is a caption';
const captionValueContainingLink =
  'This is a caption [with a link](https://bbc.com/news)';
const copyrightText = 'Getty Images';

const serviceContextStubNews = {
  imageCaptionOffscreenText: 'Image caption, ',
};

const generateFixtureData = (caption, copyright) => (
  <ServiceContext.Provider value={serviceContextStubNews}>
    <FigureContainer
      src={imageSrc}
      alt={imageAlt}
      ratio={imageRatio}
      caption={caption}
      copyright={copyright}
    />
  </ServiceContext.Provider>
);

export const FigureImage = generateFixtureData();

export const FigureImageWithCaption = generateFixtureData(captionValue);

export const FigureImageWithCaptionContainingLink = generateFixtureData(
  captionValueContainingLink,
);

export const FigureImageWithCopyright = generateFixtureData(
  null,
  copyrightText,
);

export const FigureImageWithCopyrightAndCaption = generateFixtureData(
  captionValue,
  copyrightText,
);
