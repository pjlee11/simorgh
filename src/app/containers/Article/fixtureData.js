import { blockContainingText, singleTextBlock } from '../../models/blocks';

const articleDataBuilder = (
  id,
  passportLanguage,
  home,
  headlineText,
  paragraphText,
  seoHeadline,
  promoHeadline,
  summary,
) => ({
  metadata: {
    id: `urn:bbc:ares::article:${id}`,
    locators: {
      optimoUrn: `urn:bbc:optimo:asset:${id}`,
    },
    type: 'article',
    createdBy: '',
    created: 1514808060000,
    firstPublished: 1514808060000,
    lastPublished: 1514811600000,
    lastUpdated: 1514811600000,
    passport: {
      language: passportLanguage,
      home,
      category: 'news',
      genre: null,
    },
    tags: {},
  },
  content: {
    model: {
      blocks: [
        blockContainingText('headline', headlineText),
        singleTextBlock(paragraphText),
      ],
    },
  },
  promo: {
    id: `urn:bbc:ares::article:${id}`,
    headlines: {
      seoHeadline,
      promoHeadline,
    },
    locators: {
      optimoUrn: `urn:bbc:optimo:asset:${id}`,
    },
    summary,
    timestamp: 1514811600000,
  },
});

export const articleDataNews = articleDataBuilder(
  'c0000000001o',
  'en-gb',
  'http://www.bbc.co.uk/ontologies/passport/home/News',
  'Article Headline',
  'A paragraph.',
  'Article Headline for SEO',
  'Article Headline for Promo',
  'Article summary.',
);

export const articleDataPersian = articleDataBuilder(
  'cwv2xv848j5o',
  'fa',
  'http://www.bbc.co.uk/ontologies/passport/home/Persian',
  'سرصفحه مقاله',
  'یک پاراگراف.',
  'سرصفحه مقاله',
  'سرصفحه مقاله برای ارتقاء',
  'خلاصه مقاله',
);
