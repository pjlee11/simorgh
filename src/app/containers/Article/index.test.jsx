import React from 'react';
import { shouldShallowMatchSnapshot } from '../../helpers/tests/testHelpers';
import ArticleContainer from './index';
import { articleDataNews, articleDataPersian } from './fixtureData';

// explicitly ignore console.log errors for Article/index:getInitialProps() error logging
global.console.log = jest.fn();

describe('ArticleContainer', () => {
  const newsData = { data: articleDataNews, isAmp: false, service: 'news' };

  // temporary: will be removed with https://github.com/BBC-News/simorgh/issues/836
  const newsDataNoHeadline = JSON.parse(JSON.stringify(newsData));
  newsDataNoHeadline.data.content.model.blocks.shift();

  const persianData = {
    data: articleDataPersian,
    isAmp: false,
    service: 'persian',
  };

  describe('Component', () => {
    shouldShallowMatchSnapshot(
      'should render correctly',
      <ArticleContainer data={newsData} />,
    );

    shouldShallowMatchSnapshot(
      'should render Persian article correctly',
      <ArticleContainer data={persianData} />,
    );

    shouldShallowMatchSnapshot(
      'should render null if no headline block',
      <ArticleContainer data={newsDataNoHeadline} />,
    );

    describe('no data', () => {
      shouldShallowMatchSnapshot(
        'should render correctly',
        <ArticleContainer />,
      );
    });
  });
});
