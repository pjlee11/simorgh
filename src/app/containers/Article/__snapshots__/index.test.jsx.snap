// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ArticleContainer Component no data should render correctly 1`] = `null`;

exports[`ArticleContainer Component should render Persian article correctly 1`] = `
<React.Fragment>
  <ServiceContextProvider
    service="persian"
  >
    <Header />
    <MetadataContainer
      isAmp={false}
      metadata={
        Object {
          "created": 1514808060000,
          "createdBy": "",
          "firstPublished": 1514808060000,
          "id": "urn:bbc:ares::article:cwv2xv848j5o",
          "lastPublished": 1514811600000,
          "lastUpdated": 1514811600000,
          "locators": Object {
            "optimoUrn": "urn:bbc:optimo:asset:cwv2xv848j5o",
          },
          "passport": Object {
            "category": "news",
            "genre": null,
            "home": "http://www.bbc.co.uk/ontologies/passport/home/Persian",
            "language": "fa",
          },
          "tags": Object {},
          "type": "article",
        }
      }
      promo={
        Object {
          "headlines": Object {
            "promoHeadline": "سرصفحه مقاله برای ارتقاء",
            "seoHeadline": "سرصفحه مقاله",
          },
          "id": "urn:bbc:ares::article:cwv2xv848j5o",
          "locators": Object {
            "optimoUrn": "urn:bbc:optimo:asset:cwv2xv848j5o",
          },
          "summary": "خلاصه مقاله",
          "timestamp": 1514811600000,
        }
      }
      service="persian"
    />
    <main
      role="main"
    >
      <Wrapper>
        <GridItemConstrained>
          <Blocks
            blocks={
              Array [
                Object {
                  "model": Object {
                    "blocks": Array [
                      Object {
                        "model": Object {
                          "blocks": Array [
                            Object {
                              "model": Object {
                                "text": "سرصفحه مقاله",
                              },
                              "type": "paragraph",
                            },
                          ],
                        },
                        "type": "text",
                      },
                    ],
                  },
                  "type": "headline",
                },
              ]
            }
            componentsToRender={
              Object {
                "headline": [Function],
              }
            }
          />
          <Timestamp
            timestamp={1514811600000}
          />
        </GridItemConstrained>
      </Wrapper>
      <OatWrapper>
        <GridItemConstrained>
          <Blocks
            blocks={
              Array [
                Object {
                  "model": Object {
                    "blocks": Array [
                      Object {
                        "model": Object {
                          "text": "یک پاراگراف.",
                        },
                        "type": "paragraph",
                      },
                    ],
                  },
                  "type": "text",
                },
              ]
            }
            componentsToRender={
              Object {
                "image": [Function],
                "subheadline": [Function],
                "text": [Function],
              }
            }
          />
        </GridItemConstrained>
      </OatWrapper>
    </main>
    <FooterContainer />
  </ServiceContextProvider>
</React.Fragment>
`;

exports[`ArticleContainer Component should render correctly 1`] = `
<React.Fragment>
  <ServiceContextProvider
    service="news"
  >
    <Header />
    <MetadataContainer
      isAmp={false}
      metadata={
        Object {
          "created": 1514808060000,
          "createdBy": "",
          "firstPublished": 1514808060000,
          "id": "urn:bbc:ares::article:c0000000001o",
          "lastPublished": 1514811600000,
          "lastUpdated": 1514811600000,
          "locators": Object {
            "optimoUrn": "urn:bbc:optimo:asset:c0000000001o",
          },
          "passport": Object {
            "category": "news",
            "genre": null,
            "home": "http://www.bbc.co.uk/ontologies/passport/home/News",
            "language": "en-gb",
          },
          "tags": Object {},
          "type": "article",
        }
      }
      promo={
        Object {
          "headlines": Object {
            "promoHeadline": "Article Headline for Promo",
            "seoHeadline": "Article Headline for SEO",
          },
          "id": "urn:bbc:ares::article:c0000000001o",
          "locators": Object {
            "optimoUrn": "urn:bbc:optimo:asset:c0000000001o",
          },
          "summary": "Article summary.",
          "timestamp": 1514811600000,
        }
      }
      service="news"
    />
    <main
      role="main"
    >
      <Wrapper>
        <GridItemConstrained>
          <Blocks
            blocks={
              Array [
                Object {
                  "model": Object {
                    "blocks": Array [
                      Object {
                        "model": Object {
                          "blocks": Array [
                            Object {
                              "model": Object {
                                "text": "Article Headline",
                              },
                              "type": "paragraph",
                            },
                          ],
                        },
                        "type": "text",
                      },
                    ],
                  },
                  "type": "headline",
                },
              ]
            }
            componentsToRender={
              Object {
                "headline": [Function],
              }
            }
          />
          <Timestamp
            timestamp={1514811600000}
          />
        </GridItemConstrained>
      </Wrapper>
      <OatWrapper>
        <GridItemConstrained>
          <Blocks
            blocks={
              Array [
                Object {
                  "model": Object {
                    "blocks": Array [
                      Object {
                        "model": Object {
                          "text": "A paragraph.",
                        },
                        "type": "paragraph",
                      },
                    ],
                  },
                  "type": "text",
                },
              ]
            }
            componentsToRender={
              Object {
                "image": [Function],
                "subheadline": [Function],
                "text": [Function],
              }
            }
          />
        </GridItemConstrained>
      </OatWrapper>
    </main>
    <FooterContainer />
  </ServiceContextProvider>
</React.Fragment>
`;

exports[`ArticleContainer Component should render null if no headline block 1`] = `null`;
